doc/hardware/porting/board_porting.rst

```diff
@@ -8,6 +8,65 @@ directory* with various files in it. Files in the board directory inherit
+- Support for multi-SoC boards
+- Support for reusing the SoC and board Kconfig trees outside of the Zephyr
+  build system
+- Support for advanced use cases with :ref:`sysbuild`
+- Removal of all existing arbitrary and inconsistent uses of Kconfig and folder
+  names
+
+All the documentation in this page refers to the current hardware model. Please
+refer to the documentation in Zephyr v3.6.0 (or earlier) for information on the
+previous, now obsolete, hardware model.
+
+More information about the rationale, development and concepts behind the new
+model can be found in the `original issue`_, the `original Pull Request`_ and,
+for a complete set of changes introduced, the `hardware model v2 commit`_.
+
+Some non-critial features, enhancements and improvements of the new hardware
+model are still in development. Check the `hardware model v2 enhancements issue`_
+for a complete list.
+
+The transition from the previous hardware model to the current one (commonly
+referred to as "hardware model v2") requires modifications to all existing board
+and SoC definitions. A decision was made not to provide direct backwards
+compatibility for the previous model, which leaves users transitioning from a
+previous version of Zephyr to one including the new model (v3.7.0 and onwards)
+with two options if they have an out-of-tree board (or SoC):
+
+#. Convert the out-of-tree board to the current hardware model (recommended)
+#. Take the SoC definition from Zephyr v3.6.0 and copy it to your downstream
+   repository (ensuring that the build system can find it via a
+   :ref:`zephyr module <modules>` or ``SOC_ROOT``). This will allow your board,
+   defined in the previous hardware model, to continue to work
+
+When converting your board from the previous to the current hardware model, we
+recommend first reading through this page to understand the model in detail. You
+can then use the `example-application conversion Pull Request`_ as an example on
+how to port a simple board. Additionally, a `conversion script`_ is available
+and works reliably in many cases (though multi-core SoCs may not be handled
+entirely). Finally, the `hardware model v2 commit`_ contains the full conversion
+of all existing boards from the old to the current model, so you can use it as a
+complete conversion reference.
+
+.. _original issue: https://github.com/zephyrproject-rtos/zephyr/issues/51831
+.. _original Pull Request: https://github.com/zephyrproject-rtos/zephyr/pull/50305
+.. _hardware model v2 commit: https://github.com/zephyrproject-rtos/zephyr/commit/8dc3f856229ce083c956aa301c31a23e65bd8cd8
+.. _example-application conversion Pull Request: https://github.com/zephyrproject-rtos/example-application/pull/58
+.. _conversion script: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/utils/board_v1_to_v2.py
+.. _hardware model v2 enhancements issue: https://github.com/zephyrproject-rtos/zephyr/issues/69546
```

Comment:
I would prefer if `:github:` and `:file:` were used to point to issues/PRs and files.

Reply:
Are those clickable though? do they generate links?